File: backend/src/main/java/com/poeflipfinder/backend/entity/League.java
package com.poeflipfinder.backend.entity;

/**
 * A league cached from GGG's public Leagues API. See docs/ARCHITECTURE.md
 * § League Resolution for how hasExchangeActivity is determined.
 */
public record League(
        Long id,
        String externalId,
        String displayName,
        boolean isCurrent,
        boolean hasExchangeActivity) {
}

File: app/lib/poe_flip_finder/gateways/schema/league.ex
defmodule PoeFlipFinder.Gateways.Schema.League do
  @moduledoc "Ecto mapping for the `league` table (docs/SCHEMA.md § League cache)."

  use Ecto.Schema

  schema "league" do
    field :external_id, :string
    field :display_name, :string
    field :is_current, :boolean, default: false
    field :has_exchange_activity, :boolean, default: false
    field :known_to_ggg, :boolean, default: false
  end
end
